security/pkg/nodeagent/caclient/providers/citadel/client.go

```diff
@@ -162,3 +150,44 @@ func (c *citadelClient) getTLSDialOption() (grpc.DialOption, error) {
 	transportCreds := credentials.NewTLS(&config)
 	return grpc.WithTransportCredentials(transportCreds), nil
 }
+
+func (c *citadelClient) Reconnect() error {
+	err := c.releaseResource()
+	if err != nil {
+		return fmt.Errorf("failed to close connection")
+	}
+
+	conn, err := c.buildConnection()
```

Comment:
can you refactor to share the same code with the NewCitadelClient()? Maybe all this can go to buildConnection().

Reply:
yes refactor the NewCitadelClient to use the buildconnection